src/test/java/build/buildfarm/server/AuthenticatedBuildFarmServerTest.java

```diff
@@ -0,0 +1,173 @@
+
+    @Before
+    public void setUp() throws Exception {
+        String uniqueServerName = "in-process server for " + getClass();
+
+        memoryInstanceConfig = MemoryInstanceConfig.newBuilder()
+                .setListOperationsDefaultPageSize(1024)
+                .setListOperationsMaxPageSize(16384)
+                .setTreeDefaultPageSize(1024)
+                .setTreeMaxPageSize(16384)
+                .setOperationPollTimeout(Durations.fromSeconds(10))
+                .setOperationCompletedDelay(Durations.fromSeconds(10))
+                .setCasConfig(ContentAddressableStorageConfig.newBuilder()
+                        .setMemory(MemoryCASConfig.newBuilder()
+                                .setMaxSizeBytes(640 * 1024)))
+                .setActionCacheConfig(ActionCacheConfig.newBuilder()
+                        .setDelegateCas(DelegateCASConfig.getDefaultInstance())
+                        .build())
+                .setDefaultActionTimeout(Durations.fromSeconds(600))
+                .setMaximumActionTimeout(Durations.fromSeconds(3600))
+                .build();
+
+        BuildFarmServerConfig.Builder configBuilder =
+                BuildFarmServerConfig.newBuilder()
+                        .setPort(0)
+                        .setAuth(AuthConfig.newBuilder()
+                                .setStatic(StaticAuthConfig.newBuilder()
+                                        .addTokens(AUTH_TOKEN).build()));
+        configBuilder.addInstancesBuilder()
+                .setName(INSTANCE_NAME)
+                .setDigestFunction(DigestFunction.Value.SHA256)
+                .setMemoryInstanceConfig(memoryInstanceConfig);
+
+        server = new BuildFarmServer(
+                "test",
+                InProcessServerBuilder.forName(uniqueServerName).directExecutor(),
+                configBuilder.build());
+        server.start();
+        inProcessChannel = InProcessChannelBuilder.forName(uniqueServerName)
+                .directExecutor().build();
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        inProcessChannel.shutdownNow();
+        server.stop();
+    }
+
+    @Test
+    public void listOperations() {
+        ListOperationsRequest request = ListOperationsRequest.newBuilder()
+                .setName(INSTANCE_NAME + "/operations")
+                .setPageSize(1024)
+                .build();
+
+        OperationsGrpc.OperationsBlockingStub stub1 =
+                OperationsGrpc.newBlockingStub(inProcessChannel);
+
+        try {
+            ListOperationsResponse response1 = stub1.listOperations(request);
+            fail("should have thrown");
+        } catch (Exception ex) {
+            assertThat(ex.getMessage()).containsMatch("UNAUTHENTICATED");
+        }
+
+        Metadata headers = new Metadata();
+        headers.put(StaticAuthInterceptor.AUTHORIZATION_METADATA_KEY, String.format("%s%s", StaticAuthInterceptor.AUTH_TYPE_PREFIX, AUTH_TOKEN));
```

Comment:
we should be explicit in tests and not use those constants (AUTHORIZATION_METADATA_KEY, AUTH_TYPE_PREFIX)
so this this test breaks if someone changes the value of the constant (since it will break clients...)

Reply:
Will do.

Reply:
Side note: It is case-insensitive by the way so my test with Pants worked fine.